### DISCUSSION TITLE
Are you planning to add multi-output-regression capability  to FLAML

### DISCUSSION THREAD

**Discussion by gmkyadav** [microsoft/FLAML#192] [Ideas]
Current software support single output regression. It would be helpful if you can add multi-output-regression capability

**Comment by sonichi** [comment]
Thanks for the suggestion. Could you share what's the application and what learner is currently used? It will help us prioritize.

**Reply by sonichi** [reply]
Actually, I think you can simply inherit `flaml.AutoML` class, override its fit() function, and then use `MultiOutputRegressor(ChildAutoML)`. Make sure class `ChildAutoML`  meets the requirement of `MultiOutputRegressor`. For example:

```
class ChildAutoML(flaml.AutoML):

    def fit(X_train, y_train):
         super().fit(X_train, y_train, task="regression", time_budget=your_time_budget, **your_other_kwargs)
```

**Reply by gmkyadav** [reply]
Thanks for the suggestion. Will try and let you know.

**Reply by sonichi** [reply]
@gmkyadav Does it work? The same question is asked in #277

**Reply by sonichi** [reply]
With version v0.8.0, you don't need to create a new class. See an example at: https://github.com/microsoft/FLAML/blob/85e21864ce20a323fefe297b18b8aa75f9a72f59/test/automl/test_regression.py#L198

**Reply by sonichi** [reply]
See another example using RegressorChain: #306
